crates/cargo-bsp/src/cargo_communication/request_handle.rs

```diff
@@ -1,24 +1,57 @@
 use std::io;
 use std::path::Path;
+use std::process::Command;
 
 use bsp_server::{Message, RequestId};
 use crossbeam_channel::{unbounded, Sender};
 use log::info;
 
 use bsp_types::requests::Request;
+use bsp_types::StatusCode;
 
 use crate::cargo_communication::cargo_handle::CargoHandle;
 use crate::cargo_communication::cargo_types::cargo_command::CreateCommand;
 use crate::cargo_communication::cargo_types::cargo_result::CargoResult;
 use crate::cargo_communication::cargo_types::event::Event;
 use crate::cargo_communication::request_actor::RequestActor;
+use crate::cargo_communication::request_actor_unit_graph::UnitGraphStatusCode;
 
 #[derive(Debug)]
 pub struct RequestHandle {
     cancel_sender: Sender<Event>,
     _thread: jod_thread::JoinHandle,
 }
 
+fn run_commands<R>(mut actor: RequestActor<R, CargoHandle>, requested_cmd: Command)
+where
+    R: Request + 'static,
+    R::Params: CreateCommand + Send,
+    R::Result: CargoResult,
+{
+    let unit_graph_status_code = actor.run_unit_graph();
+    // We don't run requested command, if request was cancelled during
+    // unit graph command.
+    if unit_graph_status_code == UnitGraphStatusCode::Ok {
```

Comment:
maze zrobić to porównanie z `let` żeby nie musieć derivować PartialEq dla `UnitGraphStatusCode`
`if let UnitGraphStatusCode::Ok = unit_graph_status_code`?

Reply:
Poprawione